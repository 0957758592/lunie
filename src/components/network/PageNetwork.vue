<template>
  <TmPage data-title="Network" class="small" hide-header>
    <template>
      <h3>Mainnets</h3>
      <ul>
        <li
          v-for="network in mainNets"
          :key="network.chain_id"
          class="select-network-item"
          @click="selectNetworkHandler(network)"
        >
          <NetworkItem :network="network" />
        </li>
      </ul>
      <h3>Testnets</h3>
      <ul>
        <li
          v-for="network in testNets"
          :key="network.chain_id"
          class="select-network-item"
          @click="selectNetworkHandler(network)"
        >
          <NetworkItem :network="network" />
        </li>
      </ul>
    </template>
  </TmPage>
</template>

<script>
import { Networks, NetworksResult } from "src/gql"
import NetworkItem from "./NetworkItem"

import TmPage from "common/TmPage"
export default {
  name: `page-network`,
  components: {
    TmPage,
    NetworkItem
  },
  data: () => ({
    networks: []
  }),
  computed: {
    mainNets() {
      return this.networks.filter(network => !network.testnet)
    },
    testNets() {
      return this.networks.filter(network => network.testnet)
    }
  },
  methods: {
    selectNetworkHandler(network) {
      console.log("network", network)
      this.$store.dispatch(`setNetwork`, network)
    }
  },
  apollo: {
    networks: {
      query: Networks,
      update: NetworksResult
    }
  }
}
</script>
<style scoped>
.page-profile__title {
  color: var(--bright);
  font-size: var(--h1);
  line-height: 2.25rem;
  font-weight: 500;
  padding: 1rem;
}

.page-profile__section-title {
  color: var(--txt);
}

.tm-data-msg {
  margin: 0;
}

.page-profile__section {
  padding-top: 2rem;
}

h3 {
  margin-bottom: 1rem;
  color: var(--text);
  font-size: var(--h4);
  line-height: 2.25rem;
  font-weight: 500;
}

.select-network-item {
  cursor: pointer;
  padding: 0;
  margin-bottom: 1rem;
}
</style>
