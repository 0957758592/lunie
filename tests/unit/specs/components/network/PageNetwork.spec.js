import { shallowMount, createLocalVue } from "@vue/test-utils"
import PageNetwork from "network/PageNetwork"

const localVue = createLocalVue()

describe(`PageNetwork`, () => {
  let wrapper

  const state = {
    networks: {
      networks: [
        {
          id: "gaia-testnet",
          chain_id: "gaia-123",
          logo_url: "cosmos-logo.png",
          testnet: true,
          title: "Cosmos Hub Test"
        },
        {
          id: "cosmoshub",
          chain_id: "cosmoshub",
          logo_url: "cosmos-logo.png",
          testnet: false,
          title: "Cosmos Hub"
        }
      ]
    }
  }

  beforeEach(() => {
    wrapper = shallowMount(PageNetwork, {
      localVue,
      mocks: {
        $store: {
          state,
          dispatch: jest.fn()
        },
        $route: {
          params: { height: `100` }
        },
        $router: {
          push: jest.fn()
        }
      },
      stubs: [`router-link`]
    })
  })

  it(`shows a page with a selection of networks`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has 2 network lists`, () => {
    expect(wrapper.findAll("ul").length).toBe(2)
  })

  it("sets new network ehen clicking list item", () => {
    wrapper.find(".select-network-item").trigger("click")
    expect(wrapper.vm.$store.dispatch).toHaveBeenCalledWith(
      `loadNetwork`,
      "cosmoshub"
    )
  })
})
