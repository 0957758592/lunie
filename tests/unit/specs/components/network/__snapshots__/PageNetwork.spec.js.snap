// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNetwork shows a page with a selection of networks 1`] = `
<tmpage-stub
  class="small"
  data-title="Network"
  hideheader="true"
  subtitle=""
  title=""
>
  <h3>
    Mainnets
  </h3>
   
  <ul>
    <li
      class="select-network-item"
    >
      <networkitem-stub
        network="[object Object]"
      />
    </li>
  </ul>
   
  <h3>
    Testnets
  </h3>
   
  <ul>
    <li
      class="select-network-item"
    >
      <networkitem-stub
        network="[object Object]"
      />
    </li>
  </ul>
</tmpage-stub>
`;
